File: app/src/main/java/com/widyatama/nurseassistant/view/activity/detailTodo/DetailTodoActivity.java
package com.widyatama.nurseassistant.view.activity.detailTodo;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.getbase.floatingactionbutton.FloatingActionButton;
import com.getbase.floatingactionbutton.FloatingActionsMenu;
import com.widyatama.core.base.BaseActivity;
import com.widyatama.nurseassistant.R;
import com.widyatama.nurseassistant.adapter.recyclerView.ListDetailPasienAdapter;
import com.widyatama.nurseassistant.data.model.Pasien;
import com.widyatama.nurseassistant.features.main.MainActivity;
import com.widyatama.nurseassistant.view.fragment.DoneFragment;
import com.widyatama.nurseassistant.view.fragment.SheetCallback;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class DetailTodoActivity extends BaseActivity implements DetailTodoViewContract, SheetCallback {

    private DetailTodoPresenter presenter;
    private ListDetailPasienAdapter adapter;
    private List<String> todoList = new ArrayList<>();
    private DoneFragment doneFragment;

    @Override
    protected void onInitializedView(Bundle savedInstanceState) {
        presenter = new DetailTodoPresenter(getContext(), this);
        initComponent();
        initRecycView();

    }

    private void initComponent() {
        int data = dataReceived.getInt("id");
        presenter.getList(data);

        FloatingActionButton fabMenu = (FloatingActionButton) findViewById(R.id.fabMenu);
        fabMenu.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                doneFragment = new DoneFragment(DetailTodoActivity.this);
                doneFragment.show(getSupportFragmentManager(), doneFragment.getTag());
            }
        });

        ImageView btnKembali = (ImageView) findViewById(R.id.btnKembali);
        btnKembali.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onBackPressed();
            }
        });
    }

    private void initRecycView() {
        adapter = new ListDetailPasienAdapter(todoList, getContext());
        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.rv_list);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setAdapter(adapter);
    }

    @Override
    protected Integer setLayout() {
        return R.layout.activity_detail_todo;
    }

    @Override
    public void showResult(@NotNull Pasien pasien) {
        todoList.addAll(pasien.getTodoList());
        adapter.setItem(todoList);
        adapter.notifyDataSetChanged();

        TextView name = (TextView) findViewById(R.id.name);
        TextView floor = (TextView) findViewById(R.id.floor);

        name.setText(pasien.getName());
        floor.setText(pasien.getFloor() + " " + pasien.getRoom() + " " + pasien.getBed());

    }

    @Override
    public void onDetachView() {

    }

    @Override
    public void showLoading() {

    }

    @Override
    public void hideLoading() {

    }

    @Override
    public void onFinish() {
        doneFragment.dismiss();
        presenter.delete(dataReceived.getInt("id"));
        startActivity(new Intent(this, MainActivity.class));
        finish();
    }
}

File: app/src/main/java/com/widyatama/nurseassistant/view/activity/detailTodo/DetailTodoActivity.kt
package com.widyatama.nurseassistant.view.activity.detailTodo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import com.widyatama.core.base.BaseActivity
import com.widyatama.core.extension.launchActivity
import com.widyatama.nurseassistant.R
import com.widyatama.nurseassistant.adapter.recycleview.TodoRVdapter
import com.widyatama.nurseassistant.data.model.Pasien
import com.widyatama.nurseassistant.features.main.MainActivity
import com.widyatama.nurseassistant.view.activity.listPasien.ListPasienActivity
import com.widyatama.nurseassistant.view.fragment.BottomAddPasienFragment
import com.widyatama.nurseassistant.view.fragment.DoneFragment
import com.widyatama.nurseassistant.view.fragment.SheetCallback
import kotlinx.android.synthetic.main.activity_detail_todo.*
import org.koin.android.ext.android.inject

class DetailTodoActivity : BaseActivity(), DetailTodoViewContract, SheetCallback {

    val presenter : DetailTodoPresenter<DetailTodoViewContract> by inject()
    val rvTodo : TodoRVdapter by inject()
    var listTodo = ArrayList<String>()
    var doneSheetFragment = DoneFragment(this)
    var id : Int = 0

    override fun onInitializedView(savedInstanceState: Bundle?) {
        setSupportActionBar(toolbar)
        presenter.onAttach(this)
        id = intent.extras.getInt("id")
        initList()
        presenter?.getList(id)

        fabMenu.setOnClickListener {
            doneSheetFragment = DoneFragment(this)
            doneSheetFragment.show(supportFragmentManager, doneSheetFragment.tag)
        }

        btnKembali.setOnClickListener {
            onBackPressed()
        }

    }

    private fun initList(){
        var layoutManager = LinearLayoutManager(this)
        rv_list.layoutManager = layoutManager
        rv_list.adapter =rvTodo
    }

    override fun showResult(pasien: Pasien) {
        pasien.todoList?.let { listTodo.addAll(it) }
        name.text = pasien.name
        floor.text = pasien.floor + " " + pasien.room + " " + pasien.bed
        println("=========== size "+ listTodo.size)
        rvTodo.setItem(listTodo)
        rvTodo.notifyDataSetChanged()
    }

    override fun setLayout(): Int {
        return R.layout.activity_detail_todo
    }

    override fun onFinish() {
        doneSheetFragment.dismiss()
        presenter?.delete(id)
        launchActivity<MainActivity>(true) {
        }    }
}
